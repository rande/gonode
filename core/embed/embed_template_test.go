// Copyright Â© 2014-2023 Thomas Rabaix <thomas.rabaix@gmail.com>.
//
// Use of this source code is governed by an MIT-style
// license that can be found in the LICENSE file.

package embed

import (
	"bytes"
	"html/template"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/rande/gonode/core/embed/fixtures"
	"github.com/stretchr/testify/assert"
)

func Test_Template_With_Custom_Path(t *testing.T) {

	embeds := NewEmbeds()
	embeds.Add("testmodule", fixtures.GetTestEmbedFS())

	data, err := embeds.ReadFile("testmodule", "templates/layout.html")

	assert.Nil(t, err)
	assert.NotNil(t, data)
	tpl, err := template.New("default").Parse(string(data))
	assert.NoError(t, err)

	var buf bytes.Buffer

	err = tpl.Execute(&buf, nil)
	assert.Nil(t, err)

	tpl = template.New("default")
	err = ConfigureTemplates(tpl, embeds)
	assert.Nil(t, err)

	err = tpl.ExecuteTemplate(&buf, "testmodule:content.html", nil)
	assert.Nil(t, err)

	snaps.MatchSnapshot(t, buf.String())
}
