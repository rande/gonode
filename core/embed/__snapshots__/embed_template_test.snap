
[Test_Template_With_Custom_Path - 1]


<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8" />
        <title></title>
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <meta http-equiv="X-UA-Compatible" content="IE=edge" />

        <link href="/static/bootstrap/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">

  </head>
    </head>
    <body>
        
<div>Hello World!</div>
<div>Template Inheritance in Go html/template</div>
 
<div>
    Powered by
    <a href="https://golang.org/">Go</a>
</div>


        <script src="/static/bootstrap/bootstrap.bundle.min.js"></script>
    </body>
</html>
 

---
